lib/actions/preload.js

```diff
@@ -178,11 +208,73 @@ Alternatively you can pass the --pin-device-to-release flag to pin only this dev
 		});
 };
 
+/**
+ * @param {import('balena-sdk').BalenaSDK} balenaSdk
+ * @param {string | number} appId
+ * @returns {Promise<import('balena-sdk').Application>}
+ */
+async function getAppWithReleases(balenaSdk, appId) {
+	return balenaSdk.models.application.get(appId, {
+		$expand: applicationExpandOptions,
+	});
+}
+
+async function prepareAndPreload(preloader, balenaSdk, options) {
```

Comment:
The body of this function was moved/refactored from the `action()` function. `then` chains were replaced with async/await.